examples/example_piecewise_target_function.py

```diff
@@ -0,0 +1,185 @@
+"""
+Example for evolutionary regression on a piecewise target function with one single genome
+=========================================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a piecewise target function.
+Demonstrates the use of if/else branching operator gene.
+"""
+
+# The docopt str is added explicitly to ensure compatibility with
+# sphinx-gallery.
+docopt_str = """
+   Usage:
+     example_piecewise_target_function.py [--max-generations=<N>]
+
+   Options:
+     -h --help
+     --max-generations=<N>  Maximum number of generations [default: 5000]
+"""
+
+import functools
+import matplotlib.pyplot as plt
+import numpy as np
+from docopt import docopt
+
+import cgp
+
+args = docopt(docopt_str)
+
+
+# %%
+# First we define the new operator node: If/ElseOperator for conditional outputs
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than 0
+     and outputs input1 if true and input2 otherwise """
+
+    _arity = 3
+
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        if inputs[0] > 0:
+            self._output = inputs[1]
+        else:
+            self._output = inputs[2]
+
+    _def_output = "x_1 if x_0 > 0 else x_2"
+    _def_numpy_output = "np.piecewise(x_0, [x_0> 0, x_0<=0], [x_1 [x_0> 0] , x_2[x_0<=0]])"
+    _def_sympy_output = "Piecewise((x_1, x_0 > 0), (x_2, x_0 <= 0))"
+    _def_torch_output = "torch.where(x_0 > 0, x_1, x_2)"
+
+
+# %%
+# We define a piecewise target function.  The function applies different
+# transformations to the input depending whether the input is less or greater
+# than zero. Thus to achieve high fitness,
+# an individual must make use of the if/else operator.
+def f_target(x):
+    return np.select([x <= 0, x > 0], [-x, x ** 2 + 1.0])
```

Comment:
```suggestion
    return np.select([x > 0, x <= 0], [x ** 2 + 1.0, -x])
```
for consistency with order in operator definition